metpy/xarray.py

```diff
@@ -475,17 +474,21 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-# If DatetimeAccessor does not have a strftime, monkey patch one in
-if not hasattr(DatetimeAccessor, 'strftime'):
-    def strftime(self, date_format):
-        """Format time as a string."""
-        import pandas as pd
-        values = self._obj.data
-        values_as_series = pd.Series(values.ravel())
-        strs = values_as_series.dt.strftime(date_format)
-        return strs.values.reshape(values.shape)
+# If DatetimeAccessor does not have a strftime (xarray <0.12.2), monkey patch one in
+try:
+    from xarray.core.accessors import DatetimeAccessor
+    if not hasattr(DatetimeAccessor, 'strftime'):
+        def strftime(self, date_format):
+            """Format time as a string."""
+            import pandas as pd
+            values = self._obj.data
+            values_as_series = pd.Series(values.ravel())
+            strs = values_as_series.dt.strftime(date_format)
+            return strs.values.reshape(values.shape)
 
     DatetimeAccessor.strftime = strftime
```

Comment:
This should be indented to be within the `if not hasattr(...` block:
```suggestion
        DatetimeAccessor.strftime = strftime
```